rusk/src/bin/main.rs

```diff
@@ -107,17 +107,18 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let db = rocksdb::Backend::create_or_open(db_path);
     let net = Kadcast::new(config.clone().kadcast.into());
 
-    let node = Node::new(net, db, rusk.clone());
+    let node = rusk::chain::RuskNode(Node::new(net, db, rusk.clone()));
 
     let mut _ws_server = None;
-    if config.ws.listen {
+    if config.http.listen {
         _ws_server = Some(
-            WsServer::bind(rusk, node.clone(), config.ws.listen_addr()).await?,
+            HttpServer::bind(rusk, node.clone(), config.http.listen_addr())
+                .await?,
         );
     }
 
     // node spawn_all is the entry point
-    if let Err(e) = node.spawn_all(service_list).await {
+    if let Err(e) = node.0.spawn_all(service_list).await {
```

Comment:
Why don't we spawn the HTTP server as a service here? Thus, we follow the architecture.

Reply:
The `spawn` method is exposed by the `Node` and only have `<N,DB,VM>`

The HTTP server require the whole `Rusk` instance, and not only the VM trait impl.

I agree with you that would be a better architectural choice, but I prefer to tackle that (if needed) in a separate issue